cnf-certification-test/platform/suite.go

```diff
@@ -209,38 +213,37 @@ func testContainersFsDiff(env *provider.TestEnvironment) {
 		fsDiffTester.RunTest(cut.UID)
 		switch fsDiffTester.GetResults() {
 		case testhelper.SUCCESS:
+			compliantObjects = append(compliantObjects, testhelper.NewContainerReportObject(cut.Namespace, cut.Podname, cut.Name, "Container is not modified", true))
 			continue
 		case testhelper.FAILURE:
 			tnf.ClaimFilePrintf("%s - changed folders: %v, deleted folders: %v", cut, fsDiffTester.ChangedFolders, fsDiffTester.DeletedFolders)
-			badContainers = append(badContainers, cut.Name)
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewContainerReportObject(cut.Namespace, cut.Podname, cut.Name, "Container is modified", false).
+				AddField("ChangedFolders", strings.Join(fsDiffTester.ChangedFolders, ",")).
+				AddField("DeletedFolders", strings.Join(fsDiffTester.DeletedFolders, ",")))
+
 		case testhelper.ERROR:
 			tnf.ClaimFilePrintf("%s - error while running fs-diff: %v", cut, fsDiffTester.Error)
-			errContainers = append(errContainers, cut.Name)
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewContainerReportObject(cut.Namespace, cut.Podname, cut.Name, "Error while running fs-diff", false).AddField("Error", fsDiffTester.Error.Error()))
```

Comment:
Suggestion: I like this type of report object when there's an unexpected error during the check (e.g. command failed), where you're adding the error as a new field, but this is the only place you do it in this PR. You could do the same in other places like lines 271, 299...